### ISSUE TITLE
Bug: Parametrizing fixture with cases breaks function scope

### ISSUE THREAD

**Issue by marvinpfoertner** [smarie/python-pytest-cases#202]
The following test code should pass according to the documentation, but it doesn't.
When checking the IDs of `obj1` and `obj2`, one finds that the object are not created anew for each test function.

```python
import pytest_cases


def case_tup():
    return [], {}  # This can be reproduced with any mutable object


@pytest_cases.fixture(scope="function")
@pytest_cases.parametrize_with_cases("obj1, obj2", cases=".")
def tup(obj1, obj2):
    return obj1, obj2


def test_1(tup):
    obj1, obj2 = tup

    assert len(obj1) == 0

    obj1.append(1)


def test_2(tup):
    obj1, obj2 = tup

    assert len(obj1) == 0

    obj1.append(1)
```

Workaround
-----
The following tests pass.

```python
import pytest_cases


def case_tup():
    return [], {}


@pytest_cases.fixture(scope="function")
@pytest_cases.parametrize_with_cases("tup", cases=".")
def tup(tup):
    return tup


def test_1(tup):
    obj1, obj2 = tup

    assert len(obj1) == 0

    obj1.append(1)


def test_2(tup):
    obj1, obj2 = tup

    assert len(obj1) == 0

    obj1.append(1)
```

**Comment by marvinpfoertner** [comment]
FYI @NinaEffenberger

**Comment by marvinpfoertner** [comment]
FYI @JonathanWenger, @pnkraemer, @nathanaelbosch

**Comment by smarie** [comment]
Thanks for reporting @marvinpfoertner  and sorry for the delay in answering ! I'll investigate. Thanks for posting the workaround in the meantime :)

**Comment by smarie** [comment]
I finally found the time to look at this this morning - I was finalizing the new feature `current_cases` for version 3.5.0, sorry !

This was a cache issue: the cache for `lazy_value`s used as tuples was not taking into account the pytest request context, resulting in the leakage you could see.

I remind that this cache is unfortunately mandatory for compliance with other plugins: if another plugin or an advanced user needs to access the value through pytest internals, this should not need to trigger a new call the function (the case function in your example).

Anyway this is fixed in 3.5.0 that will ship in a few minutes, thanks again a lot for spotting this !
